### PR TITLE
[Core] Refactor screen dimensions to be dynamic.

### PR DESCRIPTION
**Pull request by alloy**
Updates screen dimensions on orientation change and emits an event.

#skip_new_tests

### CODE REVIEW

File: Pod/Classes/Core/RNDynamicScreenDimensions.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#import <React/RCTBridgeModule.h>
+#import <React/RCTEventEmitter.h>
+
+@interface RNDynamicScreenDimensions : RCTEventEmitter <RCTBridgeModule>
```
Comment (alloy) [inline Pod/Classes/Core/RNDynamicScreenDimensions.h]: @ds300 We should not be using somebody else’s namespace. I.e. this needs to be prefixed with `AR`, but we can do that in follow-up.

File: Pod/Classes/Core/RNDynamicScreenDimensions.m [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+#import <Emission/AREmission.h>
+#import <objc/runtime.h>
+
+@interface RNDynamicScreenDimensions ()
+@property (nonatomic, strong, readwrite) NSDictionary *currentDimensions;
+@property (nonatomic, readwrite) BOOL isBeingObserved;
+@end
+
+
+NSDictionary *getSafeAreaInsets()
```
Comment (alloy) [inline Pod/Classes/Core/RNDynamicScreenDimensions.m]: Marking these functions as `static` will allow the compiler to inline them. Unsure what the default is.

File: src/setupJest.ts [hunk 3]
```diff
@@ -162,3 +163,26 @@ if (process.env.ALLOW_CONSOLE_LOGS !== "true") {
...
+  }
+
+  return {
+    getCurrentScreenDimensions: () => screenDimensions,
+    ProvideScreenDimensions: ({ children }) => {
+      return React.createElement(React.Fragment, null, children)
+    },
+    useScreenDimensions: () => screenDimensions,
+  }
+})
```
Comment (alloy) [inline src/setupJest.ts]: Nice 👌

File: Pod/Classes/Core/RNDynamicScreenDimensions.m [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+     name:UIDeviceOrientationDidChangeNotification
+     object:[UIDevice currentDevice]];
+  }
+  return self;
+}
+
+-(void)dealloc
+{
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
```
Comment (alloy) [inline Pod/Classes/Core/RNDynamicScreenDimensions.m]: 👍

File: src/setupJest.ts [hunk 5]
```diff
@@ -108,7 +108,7 @@ NativeModules.ARSwitchBoardModule = {
   presentMediaPreviewController: jest.fn(),
   presentArtworksSet: jest.fn(),
 }
-NativeModules.RNDynamicScreenDimensions = {
+NativeModules.ARDynamicScreenDimensions = {
```
Comment (ds300) [inline src/setupJest.ts]: we can actually delete this

Comment (ds300) [reply src/setupJest.ts]: done

**Pull request merged**